tests/glsl/shader-subgroup-shuffle.slang

```diff
@@ -0,0 +1,65 @@
+//TEST:SIMPLE(filecheck=CHECK_GLSL):  -allow-glsl -stage compute -entry computeMain -target glsl
+//TEST:SIMPLE(filecheck=CHECK_SPV):  -allow-glsl -stage compute -entry computeMain -target spirv
+
+// not testing hlsl due to missing impl
+// TEST:SIMPLE(filecheck=CHECK_HLSL): -allow-glsl -stage compute -entry computeMain -target hlsl
+// not testing cuda due to missing impl
+// TEST:SIMPLE(filecheck=CHECK_CUDA): -allow-glsl -stage compute -entry computeMain -target cuda 
+// not testing cpp due to missing impl
+// TEST:SIMPLE(filecheck=CHECK_CPP):  -allow-glsl -stage compute -entry computeMain -target cpp
+
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl -emit-spirv-directly
+#version 430
+
+//#define TEST_when_doubles_dont_crash
+
+precision highp float;
+precision highp int;
+
+//TEST_INPUT:ubuffer(data=[9 0 0 0 0], stride=4):name=inputBuffer
+buffer MyBlockName
+{
+    uint data[];
+} inputBuffer;
+
+//TEST_INPUT:ubuffer(data=[0 0], stride=4):out,name=outputBuffer
+buffer MyBlockName2 
+{
+    uint data[];
+} outputBuffer;
+
+layout(local_size_x = 32) in;
+
+void computeMain()
+{
+    // note: cpp fails; kIROp_WaveGetActiveMask fails to be called -- test #4 will fail
+    // TODO: make more robust of a test than simple call verification
+
+    outputBuffer.data[0] = subgroupShuffle(1, 1) == 1 &&
+#ifdef TEST_when_doubles_dont_crash
+                            subgroupShuffle((double)1, 1) == 1.0 && // crashes, may break spir-v spec
+#endif
+                           subgroupShuffle((float)1, 1) == 1.0f &&
+                           subgroupShuffle((int)1, 1) == 1 &&
```

Comment:
genIType means not only int but also ivec2, ivec3, or ivec4.
You need to test the vector versions of the function.
```
subgroupShuffle(int(1));
subgroupShuffle(ivec2(1));
subgroupShuffle(ivec3(1));
subgroupShuffle(ivec4(1));
```

Reply:
I did not add any vector types support for subgroup-shuffle, subgroup-shuffle-relative, subgroup-arithmetic, subgroup-shuffle, subgroup_clustered, subgroup_quad

Reply:
and creating any reasonable test for these types is fairly cumbersome at 40 different calls which can take: matrix, vector, scalar, of sint, uint, float, double, half, bool

Reply:
You are not expected to test those functions against all of available types.
You are expected to test what the spec document requires.

To be more specific, here is an example of "subgroupSuffle".
The doc says these function signatures are required.
```
        genType  subgroupShuffle(genType value,  uint id);
        genIType subgroupShuffle(genIType value, uint id);
        genUType subgroupShuffle(genUType value, uint id);
        genBType subgroupShuffle(genBType value, uint id);
        genDType subgroupShuffle(genDType value, uint id);
```
None of these take matrix and half types.
```
"genType" means float, vector<float, 2>, vector < float, 3 >, vector < float, 4>
"genIType" means int, vector<int, 2>, vector < int, 3 >, vector < int, 4>
"genUType" means uint, vector<uint, 2>, vector < uint, 3 >, vector < uint, 4>
"genBType" means bool, vector< bool, 2>, vector < bool, 3 >, vector < bool, 4>
"genDType" means double, vector< double, 2>, vector < double, 3 >, vector < double, 4>

```
To make the test less redundant, you have to use __generic.
```
__generic<T:__BuiltinType, let N:int>
bool testSubgroupXXX()
{
     typealias genType = vector<T,N>;
     return true
        && subgroupShuffle(genType(0))
        && subgroupShuffleXor(genType(0), 0)
        ....
         ;
}

void computeMain(...)
{
    bool result = true
       && testSubgroupXXX < float, 2 >
        && testSubgroupXXX < float, 3>
        && testSubgroupXXX < float, 4>
        && testSubgroupXXX < uint, 2>
        && testSubgroupXXX < uint, 3>
        && testSubgroupXXX < uint, 4>
        ...
#if defined(Test_when_double_type_works)
        && testSubgroupXXX < double, 3>
        && testSubgroupXXX < double, 3>
        && testSubgroupXXX < double, 4>
#endif // #if defined(Test_when_double_type_works)
        ;
```
This example doesn't show how to test scalar types and you need to include them.
Please check another test file as a more detailed example.
https://github.com/shader-slang/slang/blob/master/tests/glsl-intrinsic/intrinsic-basic.slang#L257

Please make the same changes to the rest of test files too.

Also note that you should have written this test BEFORE you started implementing your code in glsl.meta.slang.